src/index/update.rs

```diff
@@ -246,11 +266,10 @@ impl IndexUpdater {
 		self.push_directory(directory)?;
 
 		// Populate subdirectories
-		for sub_directory in sub_directories {
-			self.populate_directory(Some(path), &sub_directory)?;
-		}
-
-		Ok(())
+		sub_directories
+			.into_par_iter()
+			.map(|sub_directory| self.populate_directory(Some(path), &sub_directory))
```

Comment:
Could we use `.for_each()` here instead of map and collect for more clarity? I'm also unsure how this `collect()` compacts the Iter of results into a single value.

Reply:
I think it's the same as the similar Iterator method. It propagates one of the Err results (note how you were using ? before, exiting at the first error). With my other error printing change, we should get an error if the channel was closed. It probably doesn't matter much, but it seemed more clear to exit like this than to not handle the errors.

I'll add a comment here.

Reply:
See https://doc.rust-lang.org/stable/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect, it's a less-known trick.